soccer/src/soccer/strategy/agent/position/goalie.cpp

```diff
@@ -77,6 +89,28 @@ std::optional<RobotIntent> Goalie::state_to_task(RobotIntent intent) {
         intent.motion_command =
             planning::MotionCommand{"path_target", target, face_option, ignore_ball};
         return intent;
+    } else if (latest_state_ == RECEIVING) {
+        // intercept the bal
+        rj_geometry::Point current_position =
+            world_state()->get_robot(true, robot_id_).pose.position();
+        planning::LinearMotionInstant target{current_position};
+        auto receive_intercept_cmd = planning::MotionCommand{"intercept", target};
+        intent.motion_command = receive_intercept_cmd;
+        return intent;
+    } else if (latest_state_ == PASSING) {
+        // attempt to pass the ball to the target robot
+        rj_geometry::Point target_robot_pos =
+            world_state()->get_robot(true, target_robot_id).pose.position();
+        planning::LinearMotionInstant target{target_robot_pos};
+        auto pass_kick_cmd = planning::MotionCommand{"line_kick", target};
+        intent.motion_command = pass_kick_cmd;
+        intent.shoot_mode = RobotIntent::ShootMode::KICK;
+        // NOTE: Check we can actually use break beams
+        intent.trigger_mode = RobotIntent::TriggerMode::ON_BREAK_BEAM;
+        // TODO: Adjust the kick speed based on distance
+        intent.kick_speed = 4.0;
+        intent.is_active = true;
+        return intent;
```

Comment:
related to my comments above, if there was a RoleInterface to encapsulate a Passer role, there would be less duplicate code here :smile: